examples/cancelable_action/cancelable_action/server/cancelable_action.hpp

```diff
@@ -50,19 +50,17 @@ namespace examples { namespace server
             explicit reset_id(cancelable_action& this_)
               : outer_(this_)
             {
-                {
-                    hpx::thread::id old_value =
-                        outer_.id_.exchange(hpx::this_thread::get_id());
-                    HPX_ASSERT(old_value == hpx::thread::id());
-                    HPX_UNUSED(old_value);
-                }
+                std::lock_guard<hpx::lcos::local::mutex> l(outer_.mtx_);
+                hpx::thread::id old_value = outer_.id_;
```

Comment:
Just curious, why does this now require a lock rather than just being atomic?

Reply:
I might be able to roll back this change now that `thread::id` is not reference counting anymore. The reference counted `thread::id` can't be an atomic.

Reply:
Was it not possible in the end or you just didn't get to it yet (or forget to push ;))?

Reply:
It wasn't possible as even the non-ref-counted `thread;;id` ended up being not trivially copyable. I have not investigated whether this can be achieved yet, however.

Reply:
Ok, thanks. What makes the new non-ref-counted id different from the old non-ref-counted id on master? I imagined they'd be practically the same.

Reply:
As said, I have not investigated yet. It could be the explicit move constructor/move assignment I added. We might be able to remove those again.